File: priority.cpp
#include <stdio.h>
struct items{
    int id;
    int arTime;
    int bstTime;
    int compTime;
    int tat;
    int wait;
    int remain;
    int priority;
    int check;
};
int main(){
    int n,counter=0;
    printf("Enter Number of Processes: ");
    scanf("%d",&n);
    struct items process[n],checkprocess[n],temp;
    printf("Enter Arrival time, Burst Time and Priority:\n");
    for(int i=0;i<n;i++){
        printf("For Process %d :",i+1);
        scanf("%d",&process[i].arTime);
        scanf("%d",&process[i].bstTime);
        scanf("%d",&process[i].priority);
        process[i].id=i+1;
        process[i].remain=process[i].bstTime;
        counter+=process[i].bstTime;
        process[i].check=0;
    }
    for(int i=0; i<n;i++){
        for(int j=0;j<n-1;j++){
            if(process[j].arTime>process[j+1].arTime){
            temp=process[j];
            process[j]=process[j+1];
            process[j+1]=temp;
            }
        }
    }
    int runningTime=0;
    if(runningTime<process[0].arTime){
        runningTime=process[0].arTime;
    }
    
    
    
    int i =0;
    printf("P2 ");
    for (int j = runningTime; j <=counter ; j++) { for (int k = 0; k < n; k++) {
    
    
    
    if(process[k].arTime<=runningTime && process[k].check==0){
        if(process[i].priority>process[k].priority){
            printf("P%d ",process[k].id);
            i=k;
        }
    }
    }
    runningTime+=1;
    process[i].remain-=1;
    if(process[i].remain==0){
        process[i].check=1;
        process[i].compTime=runningTime;
        process[i].tat=process[i].compTime-process[i].arTime;
        process[i].wait=process[i].tat-process[i].bstTime;
        process[i].priority=999;
        }
    }
    float avgWait=0.0,avgTat=0.0;
    for (int i = 0; i < n; i++) {
        avgTat+=process[i].tat;
        avgWait+=process[i].wait;
    }
    avgTat/=n;
    avgWait/=n;
    
    
   
    printf("\nResults:\n");
    printf("Pid\tAT\tBT\tCT\tTAT\tWT\n");
    for(i=0;i<n;i++){
        printf("%d\t%d\t%d\t%d\t%d\t%d\n",process[i].id,process[i].arTime,process[i].bstTime,process[i].compTime,process[i].tat,process[i].wait);
    }
    printf("Average TAT: %f",avgTat);
    printf("Average WT: %f",avgWait);
    return 0;
}

File: priority.java
import java.util.Scanner;
import java.util.Arrays;
public class priority
{
public static void main (String[]args)
{
Scanner input = new Scanner (System.in);
System.out.print("Enter the number of process : ");
int process = input.nextInt ();
int[] id = new int[process];
int[] AT = new int[process];
int[] BT = new int[process];
int[] CT = new int[process];
int[] TT = new int[process];
int[] WT = new int[process];
int[] prior = new int[process + 1];
float avgTAT = 0;
float avgWT = 0;
int end;
int sum = 0;
int i, j, smallest, count = 0, time;
int[] x = new int[process];
for (i = 0 ; i < process ; i ++)
{
System.out.print("Enter the Process Id for P[" + (i+1)+"]:");
id[i] = input.nextInt();
System.out.print("Enter the Arrival time for P["+ (i+1)+"] : ");
AT[i] = input.nextInt();
System.out.print("Enter the Burst time for P["+ (i+1)+"]:");
BT[i] = input.nextInt();
System.out.print("Enter the Priority for P[" + (i+1)+"] :");
prior[i] = input.nextInt();
}
int temp, temp1, temp2, temp3; 
for (i = 0; i < process; i++)
{
for (j = i + 1; j < process; j++)
{
if(AT[i] > AT[j])
{
temp = AT[i];
AT[i] = AT[j];
AT[j] = temp;
temp1 = prior[i];
prior[i] = prior[j];
prior[j] = temp1;
temp2 = BT[i];
BT[i] = BT[j];
BT[j] = temp2;
temp3 = id[i];
id[i] = id[j];
id[j] = temp3;
}
}
}
for (i = 0; i < process; i++)
{
x[i] = BT[i];
}
prior[process] = 10000;
for (time = 0; count != process; time++)
{
smallest = process;
for (i = 0; i < process; i++)
{
if (AT[i] <= time && prior[i] < prior[smallest] && BT[i] > 0)
smallest = i;
}
BT[smallest]--;
if (BT[smallest] == 0)
{
count++;
end = time + 1;
CT[smallest] = end;
WT[smallest] = end - AT[smallest] - x[smallest];
TT[smallest] = end - AT[smallest];
}
}
for (i = 0; i < process; i++)
{
avgTAT += TT[i];
avgWT += WT[i];
}
System.out.println ("ID\tAT\tBT\tCT\tTAT\tWT\tPriority");
for (i = 0; i < process; i++)
{
System.out.print (id[i] + "\t" + AT[i] + "\t" + x[i] + "\t" + CT[i]
+"\t" + TT[i] + "\t" + WT[i] + "\t" + prior[i]);
System.out.println ();
}
System.out.println ("AVG TAT : " + avgTAT/process);
System.out.println ("AVG WT : " + avgWT/process);
}
}